src/main/java/net/neoforged/neoforge/event/entity/living/LivingChangeTargetEvent.java

```diff
@@ -36,30 +37,33 @@
  */
 public class LivingChangeTargetEvent extends LivingEvent implements ICancellableEvent {
     private final ILivingTargetType targetType;
-    private final LivingEntity originalTarget;
-    private LivingEntity newTarget;
+    @Nullable
+    private final LivingEntity originalAboutToBeSetTarget;
+    @Nullable
+    private LivingEntity newAboutToBeSetTarget;
 
-    public LivingChangeTargetEvent(LivingEntity entity, LivingEntity originalTarget, ILivingTargetType targetType) {
+    public LivingChangeTargetEvent(LivingEntity entity, @Nullable LivingEntity aboutToBeSetTarget, ILivingTargetType targetType) {
         super(entity);
-        this.originalTarget = originalTarget;
-        this.newTarget = originalTarget;
+        this.originalAboutToBeSetTarget = aboutToBeSetTarget;
+        this.newAboutToBeSetTarget = aboutToBeSetTarget;
         this.targetType = targetType;
     }
 
     /**
-     * {@return the new target of this entity.}
+     * {@return the new target that this entity will begin to track.}
      */
-    public LivingEntity getNewTarget() {
-        return newTarget;
+    @Nullable
+    public LivingEntity getNewAboutToBeSetTarget() {
+        return newAboutToBeSetTarget;
     }
```

Comment:
In the future, I'd prefer breaking changes like these be made into their own separate PR from a general cleanup PR, to keep changes logically separate where possible and practical.